controller/internal/enforcer/nfqdatapath/datapath_tcp.go

```diff
@@ -384,7 +398,19 @@ func (d *Datapath) processApplicationSynAckPacket(tcpPacket *packet.Packet, cont
 	// Create TCP Option
 	tcpOptions := d.createTCPAuthenticationOption([]byte{})
 
-	tcpData, err := d.tokenAccessor.CreateSynAckPacketToken(context, &conn.Auth)
+	// We add encrypt attr in the claims header field
+	compactPKI, ok := d.secrets.(*secrets.CompactPKI)
+	if !ok {
+		zap.L().Error("Secrets does not hold compactPKI type, so type assertion failed", zap.Reflect("type", reflect.TypeOf(d.secrets)))
+		return secrets.ErrNotCompactPKI
+	}
+	claimsHeaderBytes := claimsheader.NewClaimsHeader(
+		claimsheader.OptionCompressionType(compactPKI.Compressed),
+		claimsheader.OptionDatapathVersion(d.datapathVersion),
+		claimsheader.OptionEncrypt(conn.PacketFlowPolicy.Action.Encrypted()),
+	).ToBytes()
+
+	tcpData, err := d.tokenAccessor.CreateSynAckPacketToken(context, &conn.Auth, claimsHeaderBytes)
```

Comment:
Too much cut and paste of code. All this needs to be calculated once and out of a function.